src/main/java/tech/gruppone/stalker/server/controllers/PasswordController.java

```diff
@@ -0,0 +1,69 @@
+package tech.gruppone.stalker.server.controllers;
+
+import java.io.IOException;
+import javax.mail.MessagingException;
+import lombok.Value;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import reactor.core.publisher.Mono;
+import tech.gruppone.stalker.server.configuration.SmtpMailSenderConfiguration;
+import tech.gruppone.stalker.server.exceptions.EmailErrorException;
+import tech.gruppone.stalker.server.repositories.UserRepository;
+
+@Value
+@RestController
+@Log4j2
+@RequestMapping
+public class PasswordController {
+
+  /*
+    FOLLOW THIS STEPS to work properly with this endpoint and to not get an MailAuthenticationException error:
+    Since we are using Gmail to send email, we need to enable "less secure access" on the Gmail account.
+    - Log out from all the Google accounts from the browser. Login in to the gmail account configured in application.properties.
+    - Then click on https://myaccount.google.com/lesssecureapps and turn on access for less secure apps.
+     - Alternately, if the link up doesn't work, go on https://accounts.google.com/b/0/DisplayUnlockCaptcha.
+    Now, you invoke /user/password/recovery endpoint and you'll receive an mail by GruppOne!
+  */
+
+  private UserRepository userRepository;
+
+  @Autowired private SmtpMailSenderConfiguration smtpMailSender;
```

Comment:
evitare field injection. (il modo di farlo è con le due annotazioni in cima a questo esempio:

https://github.com/GruppOne/stalker-server/blob/da5bcaeacdae6336c8a52c6348d8fdbe28c1c6f1/src/main/java/tech/gruppone/stalker/server/controllers/OrganizationsController.java#L20-L26

(che andrebbero messe su ogni controller e service che abbiamo)